File: python/101 ThreadChaining.py
import threading

def hello():
    for _ in range(11):
        print(f"Hello from thread: {threading.current_thread().name}")

t1 = threading.Thread(target=hello, name="Thread-1")
t2 = threading.Thread(target=hello, name="Thread-2")
t3 = threading.Thread(target=hello, name="Thread-3")
print("\n\n\t",t1.name,"\n\n")
print(t1.is_alive())
t1.start()
print(t1.is_alive())
t1.join()
print(t1.is_alive())

print("\n\n\t",t2.name,"\n\n")
print(t2.is_alive())
t2.start()
print(t2.is_alive())
t2.join()
print(t2.is_alive())

print("\n\n\t",t3.name,"\n\n")
print(t3.is_alive())
t3.start()
print(t3.is_alive())
t3.join()
print(t3.is_alive())

File: java/101 ThreadChaining.java
package code;

class HelloThread extends Thread {
    public HelloThread(String name) {
        super(name);
    }

    @Override
    public void run() {
        for (int i = 0; i < 11; i++) {
            System.out.println("Hello from thread: " + Thread.currentThread().getName());
        }
    }
}

public class Test {
    public static void main(String[] args) {
        HelloThread t1 = new HelloThread("Thread-1");
        HelloThread t2 = new HelloThread("Thread-2");
        HelloThread t3 = new HelloThread("Thread-3");

        System.out.println("\n\n\t" + t1.getName() + "\n\n");
        System.out.println(t1.isAlive());
        t1.start();
        System.out.println(t1.isAlive());
        try {
            t1.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println(t1.isAlive());

        System.out.println("\n\n\t" + t2.getName() + "\n\n");
        System.out.println(t2.isAlive());
        t2.start();
        System.out.println(t2.isAlive());
        try {
            t2.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println(t2.isAlive());

        System.out.println("\n\n\t" + t3.getName() + "\n\n");
        System.out.println(t3.isAlive());
        t3.start();
        System.out.println(t3.isAlive());
        try {
            t3.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println(t3.isAlive());
    }
}
